src/gui/Gui_clean_exit_TEST.cc

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include <csignal>
+#include <chrono>
+#include <exception>
+#include <ignition/common/Filesystem.hh>
+#include <ignition/utils/ExtraTestMacros.hh>
+
+#include "helpers/EnvTestFixture.hh"
+#include "ignition/gazebo/Server.hh"
+#include "ignition/gazebo/gui/Gui.hh"
+#include "ignition/gazebo/test_config.hh"  // NOLINT(build/include)
+
+using namespace ignition;
+
+/////////////////////////////////////////////////
+class GazeboDeathTest
+    : public InternalFixture<::testing::TestWithParam<const char *>>
+{
+};
+
+/////////////////////////////////////////////////
+/// \brief Start the server.
+/// \param[in] _fileName Full path to the SDFormat file to load.
+void startServer(const std::string &_fileName)
+{
+  gazebo::ServerConfig config;
+  config.SetSdfFile(_fileName);
+
+  gazebo::Server server(config);
+  EXPECT_TRUE(server.Run(true, 1, true));
+}
+
+/////////////////////////////////////////////////
+/// \brief Start the GUI.
+void startGui()
+{
+  int argc = 1;
+  char *argv = const_cast<char *>("ign-gazebo-gui");
+  EXPECT_EQ(0, gazebo::gui::runGui(argc, &argv, "", ""));
+}
+
+/////////////////////////////////////////////////
+/// \brief Start both server and GUI.
+/// \param[in] _fileName Full path to the SDFormat file to load.
+void startBoth(const std::string &_fileName)
+{
+  std::thread serverThread([&]() { startServer(_fileName); });
+  std::thread guiThread(startGui);
+  using namespace std::chrono_literals;
+  std::this_thread::sleep_for(4s);
```

Comment:
Is there a specific reason why 4 seconds was chosen here? Perhaps a comment can be left that explains why in case we need to revisit this later?

Reply:
Added a comment in https://github.com/ignitionrobotics/ign-gazebo/pull/1317/commits/48b2befd4efaed475c632058ae1783193eecdd55. I wanted to make sure the system is completely initialized before sending the `SIGTERM` so that the test is reliable.